src/momento/internal/aio/_scs_data_client.py

```diff
@@ -245,6 +270,173 @@ async def list_fetch(self, cache_name: TCacheName, list_name: TListName) -> Cach
             self._log_request_error("list_fetch", e)
             return CacheListFetch.Error(convert_error(e))
 
+    async def list_length(self, cache_name: TCacheName, list_name: TListName) -> CacheListLengthResponse:
+        try:
+            self._log_issuing_request("ListLength", {"list_name": str(list_name)})
+            _validate_cache_name(cache_name)
+            _validate_list_name(list_name)
+            request = _ListLengthRequest()
+            request.list_name = _as_bytes(list_name, "Unsupported type for list_name: ")
+            response = await self._build_stub().ListLength(
+                request,
+                metadata=make_metadata(cache_name),
+                timeout=self._default_deadline_seconds,
+            )
+            self._log_received_response("ListLength", {"list_name": str(request.list_name)})
+
+            type = response.WhichOneof("list")
+            if type == "missing":
+                return CacheListLength.Miss()
+            elif type == "found":
+                return CacheListLength.Hit(response.found.length)
+            else:
+                raise UnknownException("Unknown list field")
+        except Exception as e:
+            self._log_request_error("list_length", e)
+            return CacheListLength.Error(convert_error(e))
+
+    async def list_pop_back(self, cache_name: TCacheName, list_name: TListName) -> CacheListPopBackResponse:
+        try:
+            self._log_issuing_request("ListPopBack", {"list_name": str(list_name)})
+            _validate_cache_name(cache_name)
+            _validate_list_name(list_name)
+            request = _ListPopBackRequest()
+            request.list_name = _as_bytes(list_name, "Unsupported type for list_name: ")
+            response = await self._build_stub().ListPopBack(
+                request,
+                metadata=make_metadata(cache_name),
+                timeout=self._default_deadline_seconds,
+            )
+            self._log_received_response("ListPopBack", {"list_name": str(request.list_name)})
+
+            type = response.WhichOneof("list")
+            if type == "missing":
```

Comment:
does the generated code not provide constants for these?

if not, should we consider creating our own constants for them?  The string equality checking make me a little sad

Reply:
Nope, that's the official way to do it.
https://developers.google.com/protocol-buffers/docs/reference/python-generated#oneof

I don't think we need constants, the wire protocol will remain backwards compatible.